simplecpp.cpp

```diff
@@ -2842,6 +2842,14 @@ static bool preprocessToken(simplecpp::TokenList &output, const simplecpp::Token
     return true;
 }
 
+static void addMaybeUsedMacro(std::map<std::string, std::list<simplecpp::Location> > &macros, const simplecpp::Token *tok) {
```

Comment:
well.. in general I prefer to pass by address instead of pass by reference for output parameters. When reading the simplecpp code and you see a function call.. if there is a `&` in front of a parameter in a function call that generally means the parameter is written.

Reply:
That makes sense. I am not that used to programming in C++, so I am not really familiar with the idioms. Let me fix that.

Reply:
I completely remove the helper function, given that it becomes a one-liner by your first suggestion.